app/models/mixins/authentication_mixin.rb

```diff
@@ -488,8 +485,20 @@ def authentication_component(type, method)
   end
 
   def available_authentications
-    authentication_userid_passwords + authentication_key_pairs + authentication_tokens
+    # Some callers add authentications to the association without saving but expect
+    # to be able to call available_authentications and get the cached/in-memory
+    # records back.
+    #
+    # In order to support this behavior without compromising performance in the
+    # case where the records are persisted we have to check if we have any
+    # in-memory only records.
+    if authentications.present? && authentications.pluck(:id).any?(nil)
```

Comment:
```suggestion
    if authentications.present? && authentications.any?(&:new_record?)
```

Also, does present? check for local, or does it reach out to the database first (kind of like exists?)

Reply:
> authentications.any?(&:new_record?)

So I didn't go this route because wont this load every record which is what we're trying to avoid?

Reply:
i assumed they were already loaded because .present? was checked first?

Reply:
Ugh I meant to use `any?` not `present?`, seems this was only working because it was loading them for present